### ISSUE TITLE
 Implement Message Box in Dashboard for NGOs to Communicate with Users

### ISSUE THREAD

**Issue by codewarnab** [Innovateninjas/Paws-frontend#281]
In the dashboard, there should be a message box option for NGOs to communicate with users who have reported animals. When NGOs change the status of a report, they should have the option to add a message, which will be reflected to the user who reported the animal. This communication should be one-way only, with NGOs adding messages that users can view. The messages should be stored in the database.

### Proposed Implementation:
1. **Message Box in Dashboard:**
   - Create a dedicated section in the NGO dashboard for adding messages to users. ( probably thorough a modal )
   - Design an intuitive interface with a text box for adding messages and a button to submit.

2. **Integration with Report Status Changes:**
   - Implement functionality to trigger the message box when the NGO changes the status of a report.( it should be optional only if the ngo want to add a message it should open the message box )
   - Integrate the message box within the report details page or alongside the status change options.

3. **Storage of Messages:**
   - Store the messages in the database, associating them with the respective report and user.
   - Ensure proper data structuring to facilitate retrieval and display of messages.

### Suggestions:
- Implement real-time updates to notify users immediately when a new message is added. ( see how we are notifying the user whenever status changes thorough toast notifcation )
- Provide clear guidelines for NGOs on appropriate usage of the message box to maintain professionalism and clarity in communication. 

### Links:
- Backend Repository: [Link to Backend Repository](https://github.com/Innovateninjas/Paws-Backend)

**Comment by AMISH-KHAN** [comment]
hey @codewarnab can I work on this!

**Comment by Harshitazz** [comment]
please assign it to me @codewarnab

**Comment by Ansh101112** [comment]
@codewarnab please assign this issue to me

**Comment by codewarnab** [comment]
> hey @codewarnab can I work on this!

are you comofortable with django you can work on it end to end ?

**Comment by AMISH-KHAN** [comment]
> > hey @codewarnab can I work on this!
> 
> are you comofortable with django you can work on it end to end ?

well I have  never used django before but can try and also I will do the frontend part.

**Comment by codewarnab** [comment]
> > > hey @codewarnab can I work on this!
> > 
> > 
> > are you comofortable with django you can work on it end to end ?
> 
> well I have never used django before but can try and also I will do the frontend part.

okay if you do only front-end part after approval your pr will be stay on hold wil be merged only if some other contributor make changes accordingly on backend . if your are okay with that i can assign you the issue (frontend part alone will be a level3 issue )

**Comment by codewarnab** [comment]
@AMISH-KHAN  Assigned Best of luck

**Comment by AMISH-KHAN** [comment]
thanks @codewarnab  for assigning me this issue.
I was trying set the repo locally but I have problems doing it

While setting up the cloudinary I encounter 2 problems:

1: what should go in REACT_APP_CLOUDINARY_UPLOAD_URL it is not mentioned in COUNTRIBUTING.md
2: where should i paste the new access key 

![Screenshot 2024-05-13 174644](https://github.com/Innovateninjas/Paws-frontend/assets/123164129/f8172722-ad2b-4ed9-ab02-7dd81fee1352)

**Comment by codewarnab** [comment]
@rishicds  address this issue also update the contribution.md if required

**Comment by AMISH-KHAN** [comment]
@codewarnab @rishicds any update regarding the issue?

**Comment by rishicds** [comment]
@AMISH-KHAN 

![image](https://github.com/Innovateninjas/Paws-frontend/assets/124495375/7b6b9768-a3d2-4153-a765-8037e046f537)


Go to the programmable media page and add the API environment variable in the  REACT_APP_CLOUDINARY_UPLOAD_URL


Get access key from Upload Presets in Cloudinary Settings, create a new upload preset with these settings 
![image](https://github.com/Innovateninjas/Paws-frontend/assets/124495375/18b843a8-a2f5-4626-a376-29175a0585eb)

and paste the name in  REACT_APP_CLOUDINARY_UPLOAD_PRESET
eg here name is mnxkqfc1 so it will be REACT_APP_CLOUDINARY_UPLOAD_PRESET:mnxkqfc1

**Comment by AMISH-KHAN** [comment]
@codewarnab @rishicds 
hey I am not able to register when I set up the project locally
getting this error in browser console

![image](https://github.com/Innovateninjas/Paws-frontend/assets/123164129/4879cf31-dd56-4226-ac0f-1cad7417e547)

**Comment by rishicds** [comment]
@anirbanmajumder0  look into it

**Comment by AMISH-KHAN** [comment]
@rishicds @anirbanmajumder0  

please help with this.....

> @codewarnab @rishicds hey I am not able to register when I set up the project locally getting this error in browser console
> 
> ![image](https://private-user-images.githubusercontent.com/123164129/331875516-4879cf31-dd56-4226-ac0f-1cad7417e547.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTYzOTg5MTAsIm5iZiI6MTcxNjM5ODYxMCwicGF0aCI6Ii8xMjMxNjQxMjkvMzMxODc1NTE2LTQ4NzljZjMxLWRkNTYtNDIyNi1hYzBmLTFjYWQ3NDE3ZTU0Ny5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwNTIyJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDUyMlQxNzIzMzBaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1lZjUxNzM1MGE2Y2NhMDg3Y2FjMjQ2ZTQyZGIzMDNiMTNjMzNlZTNmN2MxZGI5Nzg0NzExMzEwYTZhODY3Y2RhJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.EWNeduqpZ86pyUAil2CnI31iikjhfHVmF534zeLbBpg)

**Comment by codewarnab** [comment]
@shilpashingnapure hey can you look into that ?

**Comment by shilpashingnapure** [comment]
I think this error is happing because in locally we are trying to access the deploy link that why 
try with this setup
with docker you will able to run backend repo easily 
To run here are steps 
  1.for generate secret key ->  python manage.py shell -c "from django.core.management.utils import get_random_secret_key; print(get_random_secret_key())" 
 2. Firebase .env
- Go to project overview -> service account -> generate new private key (it will download the json file) add the values into env
- 
  3. run this cmd once you setup all .env -> docker-compose up --build 
  4. make sure , DATABASE_HOST = 'database' put this value for this key.

@codewarnab also i have mention step in readme file but its not merged yet for both without docker & with docker how can run.

**Comment by codewarnab** [comment]
> I think this error is happing because in locally we are trying to access the deploy link that why try with this setup with docker you will able to run backend repo easily To run here are steps 1.for generate secret key -> python manage.py shell -c "from django.core.management.utils import get_random_secret_key; print(get_random_secret_key())" 2. Firebase .env
> 
>     * Go to project overview -> service account -> generate new private key (it will download the json file) add the values into env
> 
> 
>     3. run this cmd once you setup all .env -> docker-compose up --build
> 
>     4. make sure , DATABASE_HOST = 'database' put this value for this key.
> 
> 
> @codewarnab also i have mention step in readme file but its not merged yet for both without docker & with docker how can run.

yup its merged now but it would be better if its is allowed use the deployed link from local sever of frontend

**Comment by shilpashingnapure** [comment]
yes , we can do , but if in backend code in **allowed host** it will mention then it will works , but in our depoly backend code we only allowing the deploys links not local host links. that whys error is getting.
and i think , in deploy code in .env file , which are allowing the allowed host , that condition is not getting statisfy.

**Comment by codewarnab** [comment]
> yes , we can do , but if in backend code in **allowed host** it will mention then it will works , but in our depoly backend code we only allowing the deploys links not local host links. that whys error is getting. and i think , in deploy code in .env file , which are allowing the allowed host , that condition is not getting statisfy.

@rishicds  see if you can fix it

**Comment by codewarnab** [comment]
@AMISH-KHAN  you can try now without docker pull the latest changes  backend and frontend both let us know if you face any issue

**Comment by AMISH-KHAN** [comment]
@codewarnab  I am getting this error now when I run python manage.py migrate

![image](https://github.com/Innovateninjas/Paws-frontend/assets/123164129/7245df0d-1cc0-4496-8c99-84616c09b34e)

can please give me a step by step guide to run the project locally as the above conversations has made it quite confusing😅

**Comment by codewarnab** [comment]
> @codewarnab I am getting this error now when I run python manage.py migrate
> 
> ![image](https://private-user-images.githubusercontent.com/123164129/333825803-7245df0d-1cc0-4496-8c99-84616c09b34e.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTY3MDg1MzUsIm5iZiI6MTcxNjcwODIzNSwicGF0aCI6Ii8xMjMxNjQxMjkvMzMzODI1ODAzLTcyNDVkZjBkLTFjYzAtNDQ5Ni04Yzk5LTg0NjE2YzA5YjM0ZS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwNTI2JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDUyNlQwNzIzNTVaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT03ZjllZDdlZTBiYzAxN2JhOGNjM2Q5ZDZmMzRlNzRiMTQ2NjA5ZGE2YjQwNDRmNjVmMDYyOGMyOTY5MWYyMTM5JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.dKAY6uGyVkACX-fqXLADUWMt2lKbgZVNkwRmE69l3Nk)
> 
> can please give me a step by step guide to run the project locally as the above conversations has made it quite confusing😅

have you properly filled the envs ?
![image](https://github.com/Innovateninjas/Paws-frontend/assets/94959855/2c772567-32dd-4ca7-8461-34effd641537)
